src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -573,6 +575,28 @@ private bool TryClose()
     {
         Exception? error = null;
 
+        // With HTTP/3 the write-side of the stream can be aborted by the client after the server
+        // has finished reading incoming content. That means errors can happen after the Input loop
+        // has finished reading.
+        //
+        // To get notification of request aborted we register to the stream closing or complete.
+        // It will notify this type that the client has aborted the request and Kestrel will complete
+        // pipes and cancel the HttpContext.RequestAborted token.
+        _streamClosedFeature.OnClosed(static s =>
+        {
+            var stream = (Http3Stream)s!;
+
+            if (!stream.IsCompleted)
+            {
+                // An error code value other than -1 indicates a value was set and the request didn't gracefully complete.
+                var errorCode = stream._errorCodeFeature.Error;
+                if (errorCode >= 0)
```

Comment:
I know this isn't new in this code, but why is this checking for `>= 0` instead of `!= -1`?

Reply:
Probably because anything negative in an error. QUIC only allows positive error codes.